pyanaconda/modules/storage/kickstart.py

```diff
@@ -52,6 +54,34 @@ def get_device_names(specs, disks_only=False, msg="{}", lineno=None):
     return drives
 
 
+def fips_check_luks_passphrase(luks_passphrase, command_name, line_number):
+    """Is the LUKS passphrase long enough in FIPS mode?
+
+    Signal a parse error if not.
+
+    This function is meant to be called indiscriminately, it will determine itself if FIPS is on.
+
+    :param str luks_passphrase: LUKS passphrase to check
+    :param str command_name: name of the command that had the passphrase
+    :param int line_number: line number where the command is found
+    :raise KickstartParseError: When the passphrase is not long enough
+    """
+    if not luks_passphrase:
+        return
+
+    if not kernel_arguments.is_enabled("fips"):
+        return
+
+    if len(luks_passphrase) >= FIPS_PASSPHRASE_MIN_LENGTH:
+        return
+
+    raise KickstartParseError(
+        _("Passphrase given in the {} command is too short in FIPS mode. "
+          "Please use at least {} characters.").format(command_name, FIPS_PASSPHRASE_MIN_LENGTH),
```

Comment:
Martin said something like: This message does not say that the FIPS mode is on, and users can't understand that the error is conditional on that.

I think that's true.